search.go

```diff
@@ -1,175 +1,152 @@
-			return nil, jErr
-		}
-
-		source["query"] = json.RawMessage(queryJSON)
-	}
-
-	if r.Size >= 0 {
-		source["size"] = r.Size
-	}
-
-	if len(r.Sort) > 0 {
-		sorts := make([]json.RawMessage, len(r.Sort))
-		for i, s := range r.Sort {
-			sortJSON, jErr := s.ToOpenSearchJSON()
-			if jErr != nil {
-				return nil, jErr
-			}
-
-			sorts[i] = sortJSON
-		}
-
-		source["sort"] = sorts
-	}
-
-	return json.Marshal(source)
-}
-
 // AddIndices sets the index list for the request.
 func (r *SearchRequest) AddIndices(indices ...string) *SearchRequest {
 	r.Index = append(r.Index, indices...)
 	return r
 }
 
-// SetSize sets the request size, limiting the number of documents returned.
+// WithSize sets the request size, limiting the number of documents returned.
 // A negative value for size will be ignored and not included in the SearchRequest.Source.
-func (r *SearchRequest) SetSize(n int) *SearchRequest {
+func (r *SearchRequest) WithSize(n int) *SearchRequest {
 	r.Size = n
 	return r
 }
 
-// AddSort to the current list of [search.Sort]s on the request.
-func (r *SearchRequest) AddSort(sort ...*search.Sort) *SearchRequest {
+// AddSorts to the current list of [Sort]s on the request.
+func (r *SearchRequest) AddSorts(sort ...Sort) *SearchRequest {
 	r.Sort = append(r.Sort, sort...)
 	return r
 }
 
-// SetQuery to be performed by the SearchRequest.
-func (r *SearchRequest) SetQuery(q search.Query) *SearchRequest {
+// WithQuery to be performed by the SearchRequest.
+func (r *SearchRequest) WithQuery(q Query) *SearchRequest {
 	r.Query = q
 	return r
 }
 
-// Do executes the SearchRequest using the provided opensearch.Client.
-// If the request is executed successfully, then a SearchResponse will be returned.
-// An error can be returned if
+// SearchResponse is a domain model union response type across all supported OpenSearch versions.
+// Currently supported versions are:
 //
-//   - The SearchRequest source cannot be created
-//   - The source fails to be marshaled to JSON
-//   - The OpenSearch request fails to executed
-//   - The OpenSearch response cannot be parsed
-func (r *SearchRequest) Do(ctx context.Context, client *opensearch.Client) (*SearchResponse, error) {
-	bodyBytes, jErr := json.Marshal(r)
-	if jErr != nil {
-		return nil, jErr
-	}
-
-	osResp, rErr := opensearchapi.SearchRequest{
-		Index: r.Index,
-		Body:  bytes.NewReader(bodyBytes),
-	}.Do(ctx, client)
-
-	if rErr != nil {
-		return nil, rErr
-	}
-
-	var respBuf bytes.Buffer
-	if _, err := respBuf.ReadFrom(osResp.Body); err != nil {
-		return nil, err
-	}
-
-	resp := &SearchResponse{
-		StatusCode: osResp.StatusCode,
-		Header:     osResp.Header,
-	}
-
-	if err := json.Unmarshal(respBuf.Bytes(), &resp); err != nil {
-		return nil, err
-	}
-
-	return resp, nil
-}
-
-// SearchResponse wraps the functionality of [opensearchapi.Response] by supporting request parsing.
+//	-OpenSearch2
 type SearchResponse struct {
+	// StatusCode of the http request
 	StatusCode int
```

Comment:
StatusCode and Header are now part of OpenSearchResult[T] that we added on Friday: https://github.com/CrowdStrike/opensearchtools/blob/main/response.go#L9